flink-connector-kafka/src/test/java/org/apache/flink/connector/kafka/sink/KafkaRecordSerializationSchemaBuilderTest.java

```diff
@@ -86,6 +89,16 @@ public void testDoNotAllowMultipleTopicSelector() {
                 .isInstanceOf(IllegalStateException.class);
     }
 
+    @Test
+    public void testDoNotAllowMultipleHeaderProducers() {
```

Comment:
I don't think you really need this restriction for the headers.

The key / value serializer setters had the check because there were different entry points for setting the serialization schema, i.e. either via `setKafkaXSerializer` or `setKeySerializationSchema`. We don't want the user to be setting the key serializer through both entrypoints, hence the check.

This isn't a problem for the header provider setting. Same goes for the Kafka partitioner, for example - we don't check multiple sets there.